components/relevancy/src/interest.rs

```diff
@@ -0,0 +1,166 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/// List of possible interests for a domain.  Each domain would have exactly one of these.  If we
+/// can't calculate an interest based on the classifier data, then we put it in `Inconclusive`
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(u32)]
+pub enum Interest {
```

Comment:
I kind of like the idea managing this via remote settings.

It seems like we could avoid breaking changes for the most part by just adding new interests rather than overwriting existing ones.  A versioning system still would be good though, maybe we would want to do something like add parent links and create a taxonomy.

This would impact some of the data structures in the rest of the component.   Maybe [InterestVector](https://github.com/mozilla/application-services/pull/6152/files#diff-967df1177b49cb7a2c410d8b0929e110ced3c88b9506ff4f618bf99452b9bc00R89-R110) would change to be a vec of `u32` values -- one for each interest.  Or maybe it should actually be a Vec of `(interest_code, count)` tuples.  That seems better if the interest vecs are sparse.

Reply:
+1 to remote settings.

Reply:
> It seems like we could avoid breaking changes for the most part by just adding new interests rather than overwriting existing ones.

Yup, that will be ideal. Though, given that we're still at the early stage and category def changes are in flux, it will take a while for us to land on a stable category list (or even better a taxonomy) to support append-only changes. Let's think about a stopgap for now, at least, future incompatible changes won't cause any weird issues for the old Fx users. What do you think?

> Maybe [InterestVector](https://github.com/mozilla/application-services/pull/6152/files#diff-967df1177b49cb7a2c410d8b0929e110ced3c88b9506ff4f618bf99452b9bc00R89-R110) would change to be a vec of u32 values -- one for each interest. Or maybe it should actually be a Vec of (interest_code, count) tuples. That seems better if the interest vecs are sparse.

Yup, that change looks reasonable. As in interests will come from RS, we would generate the interest vectors dynamically.